.github/workflows/ci.yml

```diff
@@ -496,3 +460,181 @@ jobs:
+
+  Validate-MacOS-Artifacts:
+    needs: [Publish-Binary, Publish-Image, SetEnv]
+    environment: Testing
+    runs-on: macos-latest
+    if: startsWith(github.ref, 'refs/tags/')
+    env:
+      CORSO_VERSION: ${{ needs.SetEnv.outputs.version }}
+    steps:
+      - name: Validate x86_64 binary artifacts
+        run: |
+          set -ex
+          curl -L https://github.com/alcionai/corso/releases/download/${{ env.CORSO_VERSION }}/corso_${{ env.CORSO_VERSION }}_Darwin_x86_64.tar.gz > corso.tar.gz
+          tar -xf corso.tar.gz
+          ./corso --help
+          ./corso --version 2>&1 | grep -E "^version: ${{ env.CORSO_VERSION }}$"
+      - name: Validate arm64 binary artifacts
+        run: |
+          set -ex
+          curl -L https://github.com/alcionai/corso/releases/download/${{ env.CORSO_VERSION }}/corso_${{ env.CORSO_VERSION }}_Darwin_arm64.tar.gz > corso.tar.gz
+          tar -xf corso.tar.gz
+          # TODO: test/validate arm64 executable once we have arm64 CI
+
+  Validate-Windows-Artifacts:
+    needs: [Publish-Binary, Publish-Image, SetEnv]
+    environment: Testing
+    runs-on: windows-latest
+    if: startsWith(github.ref, 'refs/tags/')
+    env:
+      CORSO_VERSION: ${{ needs.SetEnv.outputs.version }}
+    steps:
+      - name: Validate x86_64 binary artifacts
+        run: |
+          curl -L https://github.com/alcionai/corso/releases/download/${{ env.CORSO_VERSION }}/corso_${{ env.CORSO_VERSION }}_Windows_x86_64.zip -o corso.zip
+          7z x corso.zip
+          ./corso.exe --help
+          ./corso.exe --version 2>&1 | grep -E "^version: ${{ env.CORSO_VERSION }}$"
+
+  Publish-Website-Test:
+    # Similar to Publish-Website-Prod but without the release artifact validation
+    needs: [Test-Suite-Trusted, Linting, Website-Linting, SetEnv]
+    environment: ${{ needs.SetEnv.outputs.environment }}
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    defaults:
+      run:
+        working-directory: website
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/download-artifact@master
+        name: Download website from build step
+        with:
+          name: website
+          path: website/build
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          role-session-name: integration-testing
+          aws-region: us-east-1
+
+      - name: Add rotbots.txt
+        if: github.ref == 'refs/heads/main'
+        run: |
+          printf 'User-agent: *\nDisallow: /' > build/robots.txt
+
+      - name: Push website
+        run: |
+          aws s3 sync build "s3://${{ needs.SetEnv.outputs.website-bucket }}" --delete
+
+      - name: Invalidate cloudfront
+        run: |
+          aws cloudfront create-invalidation --distribution-id ${{ needs.SetEnv.outputs.website-cfid }} --paths "/*"
```

Comment:
Since we're duplicating the behavior, this might be a candidate for a composite action.